biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java

```diff
@@ -132,4 +135,17 @@ public PrintStream getOut() {
 		return out;
 	}
 
+	public String getLastOutput() {
+		return lastOutput;
+	}
+
+	public void setLastOutput(String out) {
+		if (!"".equals(out) && out != null) {
+			out = out.replaceAll("^>.*$", "");
+
+			if (!"".equals(out) && !out.startsWith("true")) {
```

Comment:
yes , but I think we can only assume that user will not change the prompt string when they calling Agent.shell("string") , also it makes no sense that user change the prompt string cause they just want to get gogo shell result and they are not using command line and don't need to see the prompt in UI.